.github/workflows/pre-commit-ci.yml

```diff
@@ -0,0 +1,19 @@
+name: pre-commit-ci
+on:
+  pull_request:
+    branches: [master]
+jobs:
+  pre-commit-ci:
+    name: Run pre-commit checks on changed files
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
Is there a way to use the `checkout` action to perform the step that the `run` statement below performs?

Reply:
Actually, the git fetch itself is redundant when we set `fetch-depth` for checkout to 0 (unlimited). I did an investigation to see whether it would be possible to checkout only the pull_request branch, the target_branch and the commits in between, but that is actually very complicated and probably not worth the engineering effort until the git history of pycytominer becomes much much larger.

Reply:
Understood - thanks for the added details and effort towards investigating this!